lib/cstruct.ml

```diff
@@ -96,7 +96,7 @@ let create_unsafe len =
   { buffer ; len ; off = 0 }
 
 let check_bounds t len =
-  Bigarray.Array1.dim t.buffer >= len
+  len >= 0 && Bigarray.Array1.dim t.buffer >= len
```

Comment:
Maybe the public function should raise `Invalid_argument` in this case. It's all a bit of a mess. Possibly we should just redefine `+` to check for overflow in modules which use unsafe accessors.